<?xml version="1.0"?>
<launch>

  <node name="obstacle_detector_node" pkg="lidar_obstacle_detector" type="obstacle_detector_node">
    <!-- Input Topic Names -->
    <param name="lidar_points_topic"                  value="/rslidar_points"/>
    <!-- <param name="lidar_points_topic"                  value="/carla/ego_vehicle/lidar"/> -->

    <!-- Output Topic Names -->
    <param name="cloud_ground_topic"                  value="obstacle_detector/cloud_ground"/>
    <param name="cloud_clusters_topic"                value="obstacle_detector/cloud_clusters"/>
    <param name="jsk_bboxes_topic"                    value="obstacle_detector/jsk_bboxes"/>
    <param name="autoware_objects_topic"              value="obstacle_detector/objects"/>
    <!-- Parameters -->
    <param name="bbox_target_frame"                   value="rslidar"/>
  </node>

  <node name="filtering_node" pkg="lidar_obstacle_detector" type="filtering" output="screen"/>
  <!-- <node pkg="lidar_obstacle_detector" name="obstacle_data" type="obstacle.py" output="screen"/> -->
  <!-- <include file="$(find zed_wrapper)/launch/zed2i.launch" /> -->
  <!-- Dynamic Reconfigure GUI -->
  <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" output="screen" />
  <!-- Autoware Object Visualizer (if you have the `lgsvl_utils` pkg, or you can use the official Autoware one) -->
  <!-- <node ns="/obstacle_detector" pkg="lgsvl_utils" type="detected_objects_visualizer" name="autoware_objects_visualizer" output="log" /> -->
  <!-- Rviz -->
  <node type="rviz" name="rviz1" pkg="rviz" args="-d $(find lidar_obstacle_detector)/rviz/mai_city.rviz" output="log" respawn="true" />

</launch>